sonar-slang-plugin/src/main/java/com/sonarsource/slang/externalreport/detekt/DetektXmlReportReader.java

```diff
@@ -0,0 +1,150 @@
+import org.sonar.api.batch.sensor.issue.NewIssueLocation;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonarsource.analyzer.commons.ExternalRuleLoader;
+
+class DetektXmlReportReader {
+
+  private static final Logger LOG = Loggers.get(DetektXmlReportReader.class);
+
+  private static final QName CHECKSTYLE = new QName("checkstyle");
+  private static final QName FILE = new QName("file");
+  private static final QName ERROR = new QName("error");
+  private static final QName NAME = new QName("name");
+  private static final QName SOURCE = new QName("source");
+  private static final QName LINE = new QName("line");
+  private static final QName MESSAGE = new QName("message");
+
+  private static final String DETEKT_PREFIX = "detekt.";
+
+  private final SensorContext context;
+
+  private int level = 0;
+
+  @Nullable
+  private InputFile inputFile = null;
+
+  private DetektXmlReportReader(SensorContext context) {
+    this.context = context;
+  }
+
+  static void read(SensorContext context, InputStream in) throws XMLStreamException, IOException {
+    new DetektXmlReportReader(context).read(in);
+  }
+
+  private void read(InputStream in) throws XMLStreamException, IOException {
+    XMLEventReader reader = XMLInputFactory.newInstance().createXMLEventReader(in);
+    while (reader.hasNext()) {
+      XMLEvent event = reader.nextEvent();
+      if (event.isStartElement()) {
+        level++;
+        onElement(event.asStartElement());
+      } else if (event.isEndElement()) {
+        level--;
+      }
+    }
+  }
+
+  private void onElement(StartElement element) throws IOException {
+    if (level == 1 && !CHECKSTYLE.equals(element.getName())) {
+      throw new IOException("Unexpected document root '" + element.getName().getLocalPart() + "' instead of 'checkstyle'.");
+    } else if (level == 2 && FILE.equals(element.getName())) {
+      onFileElement(element);
+    } else if (level == 3 && ERROR.equals(element.getName()) && inputFile != null) {
+      onErrorElement(element);
+    }
+  }
+
+  private void onFileElement(StartElement element) {
+    String filePath = getAttributeValue(element, NAME);
+    if (filePath.isEmpty()) {
+      inputFile = null;
+      return;
+    }
+    FilePredicates predicates = context.fileSystem().predicates();
+    inputFile = context.fileSystem().inputFile(predicates.or(
+      predicates.hasAbsolutePath(filePath),
+      predicates.hasRelativePath(filePath)));
+    if (inputFile == null) {
+      LOG.warn("No input file found for {}. No detekt issues will be imported on this file.", filePath);
+    }
+  }
+
+  private void onErrorElement(StartElement element) {
+    String source = getAttributeValue(element, SOURCE);
+    String line = getAttributeValue(element, LINE);
+    String message = getAttributeValue(element, MESSAGE);
+    if (source.startsWith(DETEKT_PREFIX) && !message.isEmpty()) {
```

Comment:
@alban-auzeill you can have debug here, when `source.startsWith(DETEKT_PREFIX)` is false